src/tools/flash.c

```diff
@@ -154,7 +154,7 @@ static int get_opts(struct opts* o, int ac, char** av)
     }
 
     o->filename = av[i + 1];
-    o->addr = strtoul(av[i + 2], NULL, 16);
+    o->addr = (uint32_t) strtoul(av[i + 2], NULL, 16);
```

Comment:
this is a little evil as strtoul could return 0 and is of type `unsigned long int`